operator/watchers/k8s_service_sync.go

```diff
@@ -111,16 +102,12 @@ type ServiceSyncConfiguration interface {
 // 'shared' specifies whether only shared services are synchronized. If 'false' then all services
 // will be synchronized. For clustermesh we only need to synchronize shared services, while for
 // VM support we need to sync all the services.
-func StartSynchronizingServices(ctx context.Context, wg *sync.WaitGroup, clientset k8sClient.Clientset, services resource.Resource[*slim_corev1.Service], shared bool, cfg ServiceSyncConfiguration) {
+func StartSynchronizingServices(ctx context.Context, wg *sync.WaitGroup, services resource.Resource[*slim_corev1.Service], endpoints resource.Resource[*k8s.Endpoints], shared bool, cfg ServiceSyncConfiguration) {
 	log.Info("Starting to synchronize k8s services to kvstore")
 
-	serviceOptsModifier, err := utils.GetServiceListOptionsModifier(cfg)
-	if err != nil {
-		log.WithError(err).Fatal("Error creating service option modifier")
-	}
-
 	readyChan := make(chan struct{}, 0)
 
+	wg.Add(1)
```

Comment:
The corresponding `wg.Done()` is missing.

Reply:
Fixed.